src/components/Main.tsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext } from 'react';
+import { TodosContext } from '../TodosContext';
+import { TodoList } from './TodoList';
+
+export const Main: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    visibleTodos,
+    isToggleCheckedAll,
+    setIsToggleCheckedAll,
+  } = useContext(TodosContext);
+
+  const checkAllTodos = () => {
+    const newTodos = todos.map(todo => (
+      isToggleCheckedAll
+        ? { ...todo, completed: false }
+        : { ...todo, completed: true }
```

Comment:
```suggestion
 { ...todo, completed: !isToggleCheckedAll }
```

can be more simplify